### PR TITLE
feat: added module functionality

### PR DESCRIPTION
**Pull request by applike-ss**
## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.30"
+    }
+  }
+  required_version = ">= 1.0.0"
+}
```
Comment (mfroembgen) [inline versions.tf:9]:
```suggestion
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 4.67"
    }
  }
  required_version = ">= 1.3.0"
}
```

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+variable "subnet_ids" {
+  type        = list(string)
+  description = "IDs of the subnets that should have access to the vpc endpoint"
+}
+
+variable "vpc_endpoint_service_name" {
+  type        = string
+  description = "VPC Endpoint service name (e.g. `com.amazonaws.vpce.<region>.vpce-svc-<hash>`)"
+
```
Comment (mfroembgen) [inline variables.tf]:
```suggestion
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  service_name       = var.vpc_endpoint_service_name
+  vpc_endpoint_type  = var.vpc_endpoint_service_type
+  vpc_id             = var.vpc_id
+  subnet_ids         = var.subnet_ids
+  security_group_ids = [aws_security_group.sg.id]
+
+  tags = merge({
+    Name = module.this.id
+  }, module.this.tags)
+}
```
Comment (mfroembgen) [inline main.tf:27]:
```suggestion
  tags = module.this.tags
}
```

File: versions.tf (hunk 1 above)
Comment (applike-ss) [reply versions.tf:9]: Then i can not use it in one of our projects. Also this module does not strictly need anythings really new

File: main.tf (hunk 3 above)
Comment (applike-ss) [reply main.tf:27]: but i want it to have a name, so even in ui you have a clue what it is

File: variables.tf (hunk 2 above)
Comment (applike-ss) [reply variables.tf]: fixed

**Review by mfroembgen — APPROVED** [review]

**Comment by applike-ss** [comment]
Closed, in favour of https://github.com/terraform-aws-modules/terraform-aws-vpc/tree/master/modules/vpc-endpoints

**Pull request closed without merging**